### ISSUE TITLE
Numerical effect when producing LC

### ISSUE THREAD

**Issue by manugr788** [IvS-KULeuven/PlatoSim3#99]
Dear all,

As I have mentioned during the DPA-WG 10/10 the lightcurve built with PlatoSim outputs seems to have a numerical effect whose time frequency is exactly the time needed to move from one subpixel to another. 

```
0,5 pixel of drift * 128 subpixel during 90 days * 86400 seconds = 64 subpixels in 7776000 seconds
Therefore a one subpixel displacement last (90 * 86400) / (0,5*128) = 121 500 s, = 8,23 microHz
```

1)  Can you reproduce this simulation with PlatoSim ?
2)  Do you see the same effect in raw imagettes (for example by adding all the pixel of the imagettes) ?

It could be either a numerical effect within PlatoSim or a numerical effect coming from how our lightcurve process read the PlatoSim output.

[000011007.zip](https://github.com/IvS-KULeuven/PlatoSim3/files/523793/000011007.zip)

**Comment by JorisDeRidder** [comment]
Thanks for reporting this issue, we're currently looking into it.

**Comment by JorisDeRidder** [comment]
Could you provide us with the sigma of the Gaussian that you used to do the weighted photometry?

**Comment by manugr788** [comment]
It seems that we have used a sigma of the Gaussian = 0.75

 Here is the log of lightcurve generation :

[...]
000000:tycho27:00:PSF is 128 x 128 subpixels/pixel
000026:tycho27:00:each curve with 311000 time-sample(s)
000026:tycho27:00:images    : 10 x 10
000026:tycho27:00:imagettes : 6 x 6
000026:tycho27:00:PSF centroid (-7.41330,0.34358) subpixels from PSF centre
000026:tycho27:00:Gaussian sigma is 0.750
[...]
Emmanuel

**Comment by JorisDeRidder** [comment]
Doing the photometry by simply summing the pixels in the imagette (after background subtraction) shows a time series that is OK.

Doing the photometry using a weighted Gaussian, however, indeed shows jumps in the flux time series. These jumps occur on those time points where the star crosses a subpixel border moving from one subpixel to the next. The light curve I obtain is not the same a yours, though. I see the jumps from the very start, not only after 1 month. As expected, the jumps gradually disappear when I increase the sigma of the Gaussian to do the weighted photometry. 

It would greatly help me if I can use your weighted photometry code so that I can compare and investigate this further. Where and how could I obtain the code?

Does PIS round the position of the star to the nearest subpixel? For example, if there are 4x4 subpixels per pixel, do the true star CCD positions (row, col) always end with .0, .25, .5, or .75?

**Comment by manugr788** [comment]
Following to your comment, Réza and I have addressed this issue and we noticed that when we did the comparison between PlatoSim and PIS results, we operated Platosim with 1/128 resolution whereas PIS used a 1/256 resolution. That’s why we decided yesterday to run again PIS with a 1/128 resolution. The result (find enclosed) clearly shows that using PIS with a 1/128 resolution also leads to a numerical effect, less than using PlatoSIM though.

So, is it possible to run Platosim using 1/256 resolution ? 

I mean :

SubPixels: 256

PSF:

```
Filename: psf/June15_normal/white/256/psf_201506_V2_256.hdf5 (psf 1/256 resolution)
```

NumberOfPixels: 8

> Does PIS round the position of the star to the nearest subpixel? For example, if there are 4x4 subpixels per pixel, do the true star CCD positions (row, col) always end with .0, .25, .5, or .75? 

As far as I know, PIS doesn’t do this kind of rounding.

> It would greatly help me if I can use your weighted photometry code so that I can compare and investigate this further. Where and how could I obtain the code?

Our source code is in SVN repository : https://version-lesia.obspm.fr/repos/PLATO/algorithms/WP/326100/lightcurve/lightcurve.py

svn co  --username jderidder https://version-lesia.obspm.fr/repos/PLATO/algorithms/WP/326100/lightcurve/lightcurve.py

Here you just have access to the code but it would be very complicated to run the process because you would need others modules and input files. I hope this can help though.

Emmanuel

**Comment by JorisDeRidder** [comment]
Could you remind me from which FTP site I can download psf_201506_V2_256.hdf5 ?
In your previous comment you mentioned that you attached the 1/128 result of PIS, but I couldn't find any attachment. Could you resend the attachment?

**Comment by manugr788** [comment]
> Could you remind me from which FTP site I can download psf_201506_V2_256.hdf5 ?

wget --user=jderidder --ask-password https://dev-lesia.obspm.fr/platoalgo/grids/psf/June15_normal/white/256/psf_201506_V2_256.hdf5

> Could you resend the attachment?
> Ok, I thought Github was able to manage with email attachment like JIRA, but no. I put the attachments directly into Github interface this time.

![pis_000011007_128](https://cloud.githubusercontent.com/assets/12544881/19758989/2012d020-9c2b-11e6-92ba-f3a6aec4098e.JPG)

![pis_platosim_000011007-128_lc_after_correction](https://cloud.githubusercontent.com/assets/12544881/19758996/2626d6e6-9c2b-11e6-94a5-72705056fb32.png)

**Comment by JorisDeRidder** [comment]
I have tested PlatoSim with 256^2 subpixels/pixel instead of 128, and the numerical effects are strongly reduced. A faint remnant is still present, though. However, looking at your slides, I'm not 100% sure that the effects are completely absent from the PIS light curve either. Can you give me access to the time series (ascii) of the dark blue light curve (LC PIS) in your slides? This will allow me to quantify better the difference between the PlatoSim and the PIS light curves.

**Comment by manugr788** [comment]
Find enclosed the PIS LC data and plot of 000011007 star, for both 128 and 256 resolution.

[PIS_LC_000011007.zip](https://github.com/IvS-KULeuven/PlatoSim3/files/577737/PIS_LC_000011007.zip)

![pis_128_lc](https://cloud.githubusercontent.com/assets/12544881/20093194/0ab2ce86-a59c-11e6-932b-e6de35638c6d.JPG)
![pis_256_lc](https://cloud.githubusercontent.com/assets/12544881/20093195/0ab95b48-a59c-11e6-857c-81a786effce8.JPG)

**Comment by SaraRegibo** [comment]
Should be solved when enabling jitter smoothing for the mapped PSFs.  See pull request #241

**Comment by SaraRegibo** [comment]
This effect can be avoided by setting the parameter `IncludeJitterSmoothing` to `yes`.  Instead of placing sources in one sub-pixel, it's flux will be spread out - similar to charge diffusion - over a Gaussian kernel, here with a width of half a sub-pixel.  This, in combination with no longer rounding the sub-pixel coordinates, the jitter discontinuities should disappear.

Note that also when enabling charge diffusion (by setting the parameter `IncludeChargeDiffusion` to `yes`), the sub-pixel coordinates are no longer rounded and jitter discontinuities should disappear.

So:
* `IncludeChargeDiffusion` = `yes` -> jitter discontinuities should disappear
* `IncludeJitterSmoothing` = `yes` -> jitter discontinuities should disappear
* else -> jitter discontinuities will still be there

For clarity, these parameters reside in the blocks for the `MappedGaussian` and `MappedFromFile` in the `PSF` section of our configuration file (in the `develop` branch).

**Comment by JorisDeRidder** [comment]
I'll close this issue. If you still find discontinuities, please open another issue.
